File: js_project/client/src/epubfile/html/Chapter.js
export default class Chapter {
    bookKey
    // 章节编号
    chapterIndex
    // 渲染列表
    renderBlocks
    constructor(props) {
        this.bookKey = props && props.bookKey
        this.chapterIndex = props && props.chapterIndex
        this.renderBlocks = props && props.renderBlocks
    }
}

File: lib/src/main/java/com/chineseall/epubparser/lib/html/Chapter.kt
package com.chineseall.epubparser.lib.html

import com.chineseall.epubparser.lib.render.RenderContext

class Chapter {
    var bookKey: String? = null
    // 章节编号
    var chapterIndex: Int? = null
    // 渲染列表
    var renderBlocks: MutableList<RenderBlock>? = null

    fun render(renderContext: RenderContext) {
        renderContext.onNewPage()
        renderContext.onChapterStart(chapterIndex ?: -1)
        renderBlocks?.let { blocks ->
            for (block in blocks) {
                block.render(renderContext)
            }
        }
        renderContext.onChapterEnd(chapterIndex ?: -1)
    }
}